Source/Editor/Windows/Assets/PrefabWindow.cs

```diff
@@ -355,13 +355,22 @@ private void OnPrefabApplied(Prefab prefab, Actor instance)
         private void OnPrefabOpened()
         {
             _viewport.Prefab = _asset;
-            _viewport.UpdateGizmoMode();
+            if (Editor.ProjectCache.TryGetCustomData($"UIMode:{_asset.ID}", out bool value))
+                _viewport.SetInitialUIMode(value);
+            else
+                _viewport.SetInitialUIMode(false);
```

Comment:
Could you try using `_hasUILinked` here? So we can still utilize automatic-UI mode when opening UI widget for the first time.

Reply:
Done